adapter/src/hermes/adapter/mpiio/mpiio.cc

```diff
@@ -0,0 +1,1006 @@
+                    << " exists." << std::endl;
+          for (auto &blob_name : blob_names) {
+            auto hermes_struct = mdm->DecodeBlobNameLocal(blob_name);
+            if (((hermes_struct.second.offset_ < item.second.offset_) &&
+                 (hermes_struct.second.offset_ + hermes_struct.second.size_ >
+                  item.second.offset_))) {
+              // partially contained second half
+              hapi::Blob existing_data(item.second.offset_ -
+                                       hermes_struct.second.offset_);
+              existing.first.st_bkid->Get(item.second.blob_name_, existing_data,
+                                          ctx);
+              status = existing.first.st_bkid->Put(item.second.blob_name_,
+                                                   existing_data, ctx);
+              if (status.Failed()) {
+                LOG(ERROR) << "Put Failed on adapter." << std::endl;
+              }
+            } else if (item.second.offset_ < hermes_struct.second.offset_ &&
+                       item.second.offset_ + item.second.size_ >
+                           hermes_struct.second.offset_) {
+              // partially contained first half
+              hapi::Blob existing_data(hermes_struct.second.size_);
+              existing.first.st_bkid->Get(item.second.blob_name_, existing_data,
+                                          ctx);
+              existing_data.erase(
+                  existing_data.begin(),
+                  existing_data.begin() +
+                      (hermes_struct.second.offset_ - item.second.offset_));
+              status = existing.first.st_bkid->Put(item.second.blob_name_,
+                                                   existing_data, ctx);
+              if (status.Failed()) {
+                LOG(ERROR) << "Put Failed on adapter." << std::endl;
+              }
+            } else if (hermes_struct.second.offset_ > item.second.offset_ &&
+                       hermes_struct.second.offset_ +
+                               hermes_struct.second.size_ <
+                           item.second.offset_ + item.second.size_) {
+              // fully contained
+              status =
+                  existing.first.st_bkid->DeleteBlob(item.second.blob_name_);
+              if (status.Failed()) {
+                LOG(ERROR) << "Delete blob Failed on adapter." << std::endl;
+              }
+              existing.first.st_blobs.erase(item.second.blob_name_);
+            } else {
+              // no overlap
+            }
+          }
+        }
+      }
+    }
+    data_offset += item.first.size_;
+  }
+  existing.first.ptr += data_offset;
+  existing.first.size = existing.first.size >= existing.first.ptr
+                            ? existing.first.size
+                            : existing.first.ptr;
+  mdm->Update(fp, existing.first);
+  ret = data_offset;
+  mpi_status->count_hi_and_cancelled = 0;
+  mpi_status->count_lo = ret;
```

Comment:
These fields only exist in MPICH, and are implementation specific. Here's a [relevant comment](https://github.com/open-mpi/ompi/blob/master/ompi/include/mpi.h.in#L431) from `MPI_Status` in OpenMPI. Also see section 3.2.5 of the [MPI standard](https://www.mpi-forum.org/docs/mpi-4.0/mpi40-rc-jun-21.pdf#page=79&zoom=100,118,46). Maybe there are API functions that let you set this information?

Reply:
Created issue #202.